### DISCUSSION TITLE
[FEATURE REQUEST] Enhanced Salt-Key Event for User Actions

### DISCUSSION THREAD

**Discussion by babahcv** [saltstack/salt#67439] [Feature Requests]
It would be beneficial to include additional information regarding user actions related to accepting, rejecting, or removing keys on the salt-master, similar to the information provided in the **state.apply** event. This could enhance monitoring capabilities and facilitate the configuration of reactors per user or alerting systems.

For example, extending the salt-key event to include details such as the user responsible for the action, similar to the structure below:
```
{
    "data": {
        "result": true,
        "user": "sudo_vagrant",  
        "act": "delete", 
        "id": "minion_name", 
        "_stamp": "2024-04-02T07:52:41.006504"
   }, 
    "tag": "salt/key"
}
```
```
{
    "data": {
        "result": true,
        "user": "sudo_vagrant", 
        "act": "accept",
        "id": "minion_name",
        "_stamp": "2024-04-02T07:53:03.743339"
    },
    "tag": "salt/key"
}
```
This enhanced event structure would provide insight into the user responsible for key management actions, allowing for more granular monitoring and configuration of reaction systems based on user activity. 
Thank you!

**Comment by welcome** [comment]
Hi there! Welcome to the Salt Community! Thank you for making your first contribution. We have a lengthy process for issues and PRs. Someone from the Core Team will follow up as soon as possible. In the meantime, here’s some information that may help as you continue your Salt journey.
Please be sure to review our [Code of Conduct](https://github.com/saltstack/salt/blob/master/CODE_OF_CONDUCT.md). Also, check out some of our community resources including:

  - [Community Wiki](https://github.com/saltstack/community/wiki)
  - [Salt’s Contributor Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html)
  - [Join our Community Slack](https://via.vmw.com/salt-slack)
  - [IRC on LiberaChat](https://web.libera.chat/#salt)
  - [Salt Project YouTube channel](https://www.youtube.com/channel/UCpveTIucFx9ljGelW63-BWg)
  - [Salt Project Twitch channel](https://www.twitch.tv/saltprojectoss)

There are lots of ways to get involved in our community. Every month, there are around a dozen opportunities to meet with other contributors and the Salt Core team and collaborate in real time. The best way to keep track is by subscribing to the Salt Community Events Calendar.
If you have additional questions, email us at saltproject@vmware.com. We’re glad you’ve joined our community and look forward to doing awesome things with you!
